common/ferrostar/src/simulation.rs

```diff
@@ -125,33 +158,108 @@ pub fn advance_location_simulation(
 #[cfg(test)]
 mod tests {
     use super::*;
+    use crate::algorithms::snap_user_location_to_line;
+    use geo::HaversineDistance;
+    use rstest::rstest;
 
-    #[test]
-    fn advance_to_next_location() {
-        let mut state = location_simulation_from_coordinates(vec![
-            GeographicCoordinate { lng: 0.0, lat: 0.0 },
-            GeographicCoordinate { lng: 1.0, lat: 1.0 },
-            GeographicCoordinate { lng: 2.0, lat: 2.0 },
-            GeographicCoordinate { lng: 3.0, lat: 3.0 },
-        ])
+    #[rstest]
+    #[case(None)]
+    #[case(Some(10.0))]
+    fn advance_to_next_location(#[case] resample_distance: Option<f64>) {
+        let mut state = location_simulation_from_coordinates(
+            vec![
+                GeographicCoordinate { lng: 0.0, lat: 0.0 },
+                GeographicCoordinate {
+                    lng: 0.0001,
+                    lat: 0.0001,
+                },
+                GeographicCoordinate {
+                    lng: 0.0002,
+                    lat: 0.0002,
+                },
+                GeographicCoordinate {
+                    lng: 0.0003,
+                    lat: 0.0003,
+                },
+            ],
+            resample_distance,
+        )
         .expect("Unable to initialize simulation");
 
+        // Loop until state no longer changes
         let mut states = vec![state.clone()];
-        for _ in 0..4 {
-            state = advance_location_simulation(&state, SimulationAdvanceStyle::JumpToNextLocation);
+        loop {
+            let new_state = advance_location_simulation(&state);
+            if new_state == state {
+                break;
+            }
+            state = new_state;
             states.push(state.clone());
         }
 
-        insta::assert_yaml_snapshot!(states);
+        insta::assert_yaml_snapshot!(format!("{:?}", resample_distance), states);
```

Comment:
This funky first argument gives the snapshot a deterministic name based on the args passed. Otherwise it's just a sequential value like 1, 2, etc., but the rstest crate ordering is not really deterministic in which order it executes cases ;)